pkg/rancher-desktop/window/index.ts

```diff
@@ -218,6 +218,57 @@ function createView() {
 
   if (currentExtension?.id) {
     webPreferences.partition = `persist:rdx-${ currentExtension.id }`;
+    const webRequest = Electron.session.fromPartition(webPreferences.partition).webRequest;
+
+    webRequest.onBeforeSendHeaders((details, callback) => {
+      const source = details.webContents?.getURL() ?? '';
+      const requestHeaders = { ...details.requestHeaders };
+
+      if (isInternalURL(source)) {
+        // If the request is coming from the extension, remove the Origin: header
+        // because it has x-rd-extension:// nonsense (relative to the server).
+        delete requestHeaders.Origin;
+      }
+      callback({ requestHeaders });
+    });
+
+    webRequest.onHeadersReceived((details, callback) => {
+      const sourceURL = details.webContents?.getURL() ?? '';
+
+      if (!isInternalURL(sourceURL)) {
+        // Do not rewrite requests from outside the extension (e.g. iframe).
+        callback({});
+
+        return;
+      }
+
+      // Insert (or overwrite) CORS headers to pretend this was allowed.  While
+      // ideally we can just disable `webSecurity` instead, that seems to break
+      // the preload script (which breaks the extension APIs).
+      const responseHeaders: Record<string, string|string[]> = { ...details.responseHeaders };
+      const { compare } = new Intl.Collator('en', { sensitivity: 'accent' });
+      const overwriteHeaders = [
+        'Access-Control-Allow-Headers',
+        'Access-Control-Allow-Methods',
+        'Access-Control-Allow-Origin',
+      ];
+
+      for (const header of overwriteHeaders) {
+        const match = Object.keys(responseHeaders).find(k => compare(header, k) === 0);
+
+        responseHeaders[match ?? header] = '*';
+      }
+
+      if (details.method !== 'OPTIONS') {
+        // For any request that's not a CORS preflight, just overwrite the headers.
+        callback({ responseHeaders });
+      } else {
+        // For CORS preflights, also change the status code.
+        const prefix = /\s+/.exec(details.statusLine)?.shift() ?? 'HTTP/1.1';
+
+        callback({ responseHeaders, statusLine: `${ prefix } 204` });
```

Comment:
nit; would it make sense to make a constant for 204 (noContent)?

Reply:
It didn't seem like it was worth it, because it needs to be in the string anyway. I'll add the `No Content` part to the string so it's easier to read?